src/controller/postController.js

```diff
@@ -0,0 +1,138 @@
+const User = require('../database/models/userModel');
+const Post = require('../database/models/postModel');
+const uploads = require('../utils/uploadHandler');
+
+exports.createPost = [
+  async (req, res) => {
+    try {
+      const { location } = req.body;
+      const { email } = req.user;
+      const user = await User.findOne({ email });
+
+      const posts = req.files.map((file) => file.path);
+
+      const newPost = await Post.create({
+        userId: user.id,
+        media: posts,
+        location,
+        createdAt: new Date(),
+      });
+
+      res.status(201).json({
+        message: 'Post created successfully',
+        post: newPost,
+      });
+    } catch (error) {
+      res.status(500).json({
+        message: 'Could not Create Post',
+        error: error.message,
+      });
+    }
+  },
+];
+
+exports.getallPosts = async (req, res) => {
+  try {
+    const { email } = req.user;
+    const user = await User.findOne({ email });
+
+    const userPosts = await Post.find({ userId: user.id });
+
+    return res.status(200).json({
+      message: 'SUCCESS',
+      data: userPosts,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Could not find Posts',
+    });
+  }
+};
+
+exports.getPost = async (req, res) => {
+  try {
+    const { email } = req.user;
+    const user = await User.findOne({ email });
```

Comment:
No need to fetch the user again, everything you need for loggedIn user is in req.user